app/controllers/git_controller.rb

```diff
@@ -0,0 +1,37 @@
+require 'json'
+require 'git'
+require 'fileutils'
+
+class GitController < ApplicationController
+
+  def pull
+    conf = get_config
```

Comment:
Reads the file every time, should only be loaded once at startup ( or in some other way infrequently )